app/.eslintrc.js

```diff
@@ -0,0 +1,49 @@
+module.exports = {
+  root: true,
+  extends: [
+    'nava',
+    // Disable ESLint code formatting rules which conflict with Prettier
+    'prettier',
+    // `next` should be extended last according to their docs
+    // https://nextjs.org/docs/basic-features/eslint
+    'next/core-web-vitals',
+  ],
+  // Additional lint rules. These get layered onto the top-level rules.
+  overrides: [
+    // Lint config specific to Test files
+    {
+      files: ['tests/**'],
+      plugins: ['jest'],
+      extends: ['plugin:jest/recommended'],
+    },
+    // Lint config specific to TypeScript files
+    {
+      files: '**/*.+(ts|tsx)',
+      parserOptions: {
+        // These paths need defined to support rules that require type information
+        tsconfigRootDir: __dirname,
+        project: ['./tsconfig.json'],
+      },
+      extends: [
+        'plugin:@typescript-eslint/recommended',
+        // Disable vanilla ESLint rules that conflict with those in @typescript-eslint
+        'plugin:@typescript-eslint/eslint-recommended',
+        // Rules that specifically require type information
+        'plugin:@typescript-eslint/recommended-requiring-type-checking',
+      ],
+      plugins: ['@typescript-eslint'],
+      rules: {
+        // Prevent dead code accumulation
+        '@typescript-eslint/no-unused-vars': 'error',
+        // The usage of `any` defeats the purpose of typescript. Consider using `unknown` type instead instead.
```

Comment:
Totally unrelated sidebar but for the template did the team discuss the merits of js vs ts somewhere?

Reply:
@aligg Good question, I'm not sure. Do you remember @lorenyu @rocketnova?

Reply:
Don't think so, but at this point I think there's enough in house expertise with typescript and the ecosystem is mature enough that I think it's a pretty easy decision to go with typescript for the type safety which speeds up development